src/main/java/com/example/bank/registration/ClientService.java

```diff
@@ -0,0 +1,76 @@
+package com.example.bank.registration;
+
+import com.example.bank.exception.ClientNotFoundException;
+import com.example.bank.registration.jpa.Client;
+import com.example.bank.registration.jpa.ClientRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.time.Period;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ClientService {
+    private final ClientRepository clientRepository;
+    private final String clientLessThan18YearsOldMessage = "Client has to be adult.";
+
+    Client registerClient(ClientRequest clientRequest) {
+        log.info("Start client's registration.");
+        int years = Period.between(clientRequest.birthDate(), LocalDate.now())
+                .getYears();
+        if (years >= 18) {
+            Client client = new Client(clientRequest.id(), clientRequest.firstName(), clientRequest.lastName(),
+                    clientRequest.birthDate(), clientRequest.email(), clientRequest.address());
+            clientRepository.save(client);
+            log.info("Client's registration passed.");
+            return client;
+        } else {
+            log.error("Client's registration failed.");
+            throw new IllegalArgumentException(clientLessThan18YearsOldMessage);
+        }
+    }
+
+    Client getClient(Long id) {
+        Client client = clientRepository.findById(id)
+                .orElseThrow(() -> {
+                    log.error("Error. There is no client with id #" + id + " in database.");
```

Comment:
wielokrotnie powtarza sie ten string, dalbym to do stalej

Reply:
wydzielono do zmiennej